lib/api/utilities/resource_link_parser.rb

```diff
@@ -30,20 +30,65 @@
 module API
   module Utilities
     module ResourceLinkParser
-      def self.parse(resource_link)
-        ::API::Root.routes.each do |route|
-          route_options = route.instance_variable_get(:@options)
-          match = route_options[:compiled].match(resource_link)
-
-          if match
-            return {
-              ns: /\/(?<ns>\w+)\//.match(route_options[:namespace])[:ns],
-              id: match[:id]
-            }
+      # N.B. valid characters for URL path segments as of
+      # http://tools.ietf.org/html/rfc3986#section-3.3
+      SEGMENT_CHARACTER = '(\w|[-~!$&\'\(\)*+\.,:;=@]|%[0-9A-Fa-f]{2})'
+      RESOURCE_REGEX = "/api/v(?<version>\\d)/(?<namespace>\\w+)/(?<id>#{SEGMENT_CHARACTER}*)\\z"
```

Comment:
I know. However, my thought was that we said that we probably don't like nested resources like in the old days:

````
/api/v3/projects/5/work_packages/7
````

Your member example would be correct, if we would not link to a user.

However, to get my point home: The old implementation relied on a path segment called `":id"` being defined, which would also have failed (or not worked reliably) for those kinds of resources. So I am at least as good as that one ;-)

My hope is that we will never need nested links for resources we will allow setting via link.
That would also open a wide variety of new problems, e.g. setting `project/2/member/4` as member for project **3**.